src/pages/workspace/perDiem/ImportedPerDiemPage.tsx

```diff
@@ -0,0 +1,174 @@
+        let errors: Errors = {};
+
+        if (!requiredColumns.every((requiredColumn) => columns.includes(requiredColumn.value))) {
+            // eslint-disable-next-line rulesdir/prefer-early-return
+            requiredColumns.forEach((requiredColumn) => {
+                if (!columns.includes(requiredColumn.value)) {
+                    errors.required = translate('spreadsheet.fieldNotMapped', {fieldName: requiredColumn.text});
+                }
+            });
+        } else {
+            const duplicate = findDuplicate(columns);
+            const duplicateColumn = columnRoles.find((role) => role.value === duplicate);
+            if (duplicateColumn) {
+                errors.duplicates = translate('spreadsheet.singleFieldMultipleColumns', {fieldName: duplicateColumn.text});
+            } else {
+                errors = {};
+            }
+        }
+        return errors;
+    }, [requiredColumns, spreadsheet?.columns, translate, columnRoles]);
+
+    const importPerDiemRates = useCallback(() => {
+        setIsValidationEnabled(true);
+        const errors = validate();
+        if (Object.keys(errors).length > 0) {
+            return;
+        }
+
+        const columns = Object.values(spreadsheet?.columns ?? {});
+        const perDiemDestinationColumn = columns.findIndex((column) => column === CONST.CSV_IMPORT_COLUMNS.DESTINATION);
+        const perDiemSubRateColumn = columns.findIndex((column) => column === CONST.CSV_IMPORT_COLUMNS.SUBRATE);
+        const perDiemCurrencyColumn = columns.findIndex((column) => column === CONST.CSV_IMPORT_COLUMNS.CURRENCY);
+        const perDiemAmountColumn = columns.findIndex((column) => column === CONST.CSV_IMPORT_COLUMNS.AMOUNT);
+        const perDiemDestination = spreadsheet?.data[perDiemDestinationColumn].map((destination) => destination) ?? [];
+        const perDiemSubRate = spreadsheet?.data[perDiemSubRateColumn].map((subRate) => subRate) ?? [];
+        const perDiemCurrency = spreadsheet?.data[perDiemCurrencyColumn].map((currency) => sanitizeCurrencyCode(currency)) ?? [];
+        const perDiemAmount = spreadsheet?.data[perDiemAmountColumn].map((amount) => amount) ?? [];
+        const perDiemUnits = generatePerDiemUnits(
+            perDiemDestination?.slice(containsHeader ? 1 : 0),
+            perDiemSubRate?.slice(containsHeader ? 1 : 0),
+            perDiemCurrency?.slice(containsHeader ? 1 : 0),
+            perDiemAmount?.slice(containsHeader ? 1 : 0),
+        );
+
+        const rowsLength = perDiemDestination.length - (containsHeader ? 1 : 0);
+
+        if (perDiemUnits) {
+            setIsImportingPerDiemRates(true);
+            PerDiem.importPerDiemRates(policyID, perDiemCustomUnit?.customUnitID ?? '', perDiemUnits, rowsLength);
+        }
+    }, [validate, spreadsheet?.columns, spreadsheet?.data, containsHeader, policyID, perDiemCustomUnit?.customUnitID]);
+
+    const spreadsheetColumns = spreadsheet?.data;
+    if (!spreadsheetColumns) {
+        return;
```

Comment:
Deeplinking to this page shows an empty page. A `NotFound` page with suitable title and subtitle with providing a link directing the user to the import page might be better?
![Screenshot 2024-11-25 at 5 05 04 PM](https://github.com/user-attachments/assets/b8743c57-da1b-42d2-bf35-128ba3b45527)

Reply:
https://github.com/user-attachments/assets/9968bcd3-70fd-4a81-bde5-08095f872c0b